app/static/js/search.js

```diff
@@ -22,6 +26,13 @@ function handleSearchResults($ajaxPromise) {
       $resultsContainer.append(data);
 
       // Need to add event handlers for these after inserting them on the page
+
+      if ($.trim($(".abstract abstract--full").text().split(" ")).length < 500) {
+        $(".show_more_button").hide()
+        $(".show_less_button").hide()
+
+      }
+    
```

Comment:
I think what you want to do is, rather than checking the length of the abstract, is to attach the events that call showFullAbstract and hideFullAbstract to the buttons.

Then you can show and hide the buttons, as appropriate, in showFullAbstract and hideFullAbstract.